library/src/blas3/rocblas_trsm.hpp

```diff
@@ -0,0 +1,1737 @@
+                                     rocblas_stride stride_a,
+                                     V              b,
+                                     rocblas_int    ldb,
+                                     rocblas_stride stride_b,
+                                     rocblas_int    offset_a,
+                                     rocblas_int    offset_b)
+    {
+        const T* xa = load_ptr_batch(a, hipBlockIdx_z, offset_a, stride_a);
+        T*       xb = load_ptr_batch(b, hipBlockIdx_z, offset_b, stride_b);
+
+        size_t tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+        size_t ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+        if(tx < rows && ty < cols)
+            xb[tx + ldb * ty] = xa[tx + lda * ty];
+    }
+
+    /* ===============copy helper============================================= */
+    template <typename T, typename U, typename V>
+    void copy_block_unit(rocblas_handle handle,
+                         rocblas_int    m,
+                         rocblas_int    n,
+                         U              src,
+                         rocblas_int    src_ld,
+                         rocblas_stride src_stride,
+                         V              dst,
+                         rocblas_int    dst_ld,
+                         rocblas_stride dst_stride,
+                         rocblas_int    batch_count,
+                         rocblas_int    offset_src = 0,
+                         rocblas_int    offset_dst = 0)
+    {
+        rocblas_int blocksX = (m - 1) / 128 + 1; // parameters for device kernel
+        rocblas_int blocksY = (n - 1) / 8 + 1;
+        dim3        grid(blocksX, blocksY, batch_count);
+        dim3        threads(128, 8);
+
+        hipLaunchKernelGGL(copy_matrix_trsm<T>,
+                           grid,
+                           threads,
+                           0,
+                           handle->rocblas_stream,
+                           m,
+                           n,
+                           sizeof(T),
+                           src,
+                           src_ld,
+                           src_stride,
+                           dst,
+                           dst_ld,
+                           dst_stride,
+                           offset_src,
+                           offset_dst);
+    }
+
+    /* ===============left==================================================== */
+
+    template <rocblas_int BLOCK, bool BATCHED, typename T, typename U, typename V>
+    rocblas_status rocblas_trsm_left(rocblas_handle    handle,
+                                     rocblas_fill      uplo,
+                                     rocblas_operation transA,
+                                     rocblas_int       m,
+                                     rocblas_int       n,
+                                     const T*          alpha,
+                                     rocblas_stride    stride_alpha,
```

Comment:
`alpha` should be scalar. I thought we decided not to support multiple scalars.

Reply:
Only a single scalar is supported through the API, multiple scalars are supported at the template level. This similar functionality was supported in some of the L1 functions as Juan specifically requested it, although we are assured that in that case they will be only on the device, which I did not deal with in the PR at this time.

Reply:
Removed multiple scalars in 66c771e since they will be removed in gemm anyway.